> One unit test failed, but I am not sure if it is caused by this PR: https://github.com/kvcache-ai/Mooncake/actions/runs/28764086030/job/85287643208?pr=2739
> 
> @Aionw @NUABO Could you take a look?

<img width="1005" height="499" alt="image" src="https://github.com/user-attachments/assets/dec27f74-e5ae-4a2e-a0b0-0fe9c30586cc" />


> One unit test failed, but I am not sure if it is caused by this PR: https://github.com/kvcache-ai/Mooncake/actions/runs/28764086030/job/85287643208?pr=2739
> 
> @Aionw @NUABO Could you take a look?

hi @ykwd This appears to be a CI environment issue and is not related to this submission.
```
I0706 03:39:15.117377  5736 master_service.cpp:7837] client_id=13061595477301827689-5342800102356941696, action=client_expired
I0706 03:39:16.117547  5736 master_service.cpp:7837] client_id=11188675390361187660-9718843734784196502, action=client_expired
WARNING: Logging before InitGoogleLogging() is written to STDERR
I0706 03:39:18.998122  6021 client_metric.cpp:114] Client metrics enabled (default enabled)
I0706 03:39:18.998195  6021 client_metric.cpp:115] Client bandwidth summary enabled via MC_STORE_CLIENT_METRIC_BANDWIDTH
I0706 03:39:18.999593  6021 client_service.cpp:287] client_id=5857753429562426084-17843503273471751343
I0706 03:39:18.999604  6021 client_service.cpp:298] Client metrics enabled but reporting disabled (interval=0)
I0706 03:39:19.000367  5747 master_service.cpp:4842] Storage root directory or cluster ID is not set. persisting data is disabled.
I0706 03:39:19.000440  6021 client_service.cpp:929] Storage root directory is not set. persisting data is disabled.
I0706 03:39:19.000473  6021 transfer_engine_impl.cpp:886] Metrics reporting is disabled (set MC_TE_METRIC=1 to enable)
I0706 03:39:19.000480  6021 transfer_engine_impl.cpp:115] Transfer Engine parseHostNameWithPort. server_name: 0.0.0.0 port: 12795
I0706 03:39:19.000624  6021 transfer_metadata_plugin.cpp:1293] Found active interface eth0 with IP 10.1.0.154
I0706 03:39:19.000636  6021 transfer_metadata_plugin.cpp:1282] Skipping interface docker0 (not UP or not RUNNING)
I0706 03:39:19.000658  6021 transfer_engine_impl.cpp:182] Transfer Engine RPC using new RPC mapping, listening on 10.1.0.154:15239
I0706 03:39:19.002117  6021 client_service.cpp:747] Transfer engine auto discovery is disabled for protocol: tcp
E0706 03:39:19.002569  6021 transfer_metadata_plugin.cpp:292] GET http://127.0.0.1:8080/metadata?key=mooncake%2Fram%2F0.0.0.0%3A12795 http=404 body: metadata not found
W0706 03:39:19.002581  6021 transfer_metadata.cpp:929] Failed to retrieve segment descriptor, name 0.0.0.0:12795
I0706 03:39:19.002977  6021 tcp_transport.cpp:691] TcpTransport: listen on port 16509
I0706 03:39:19.003865  6021 real_client.cpp:715] Successfully created client on port 12795 after 1 attempt(s)
I0706 03:39:19.003875  6021 real_client.cpp:767] Local buffer size is 0, skip registering local memory
I0706 03:39:19.003880  6021 real_client.cpp:823] Mounting segment: 4294967296 bytes, 4294967296 of 4294967296
I0706 03:39:19.004323  5747 master_service.cpp:782] client_id=5857753429562426084-17843503273471751343, action=mount_segment, segment_name=0.0.0.0:12795
I0706 03:39:19.005048  6021 real_client.cpp:896] Starting IPC server at @mooncake_client_50052.sock
I0706 03:39:19.005152  6021 real_client.cpp:5295] start dummy_client_monitor_thread
I0706 03:39:19.005191  6021 real_client_main.cpp:132] Starting real client service on 0.0.0.0:50052
W0706 03:39:19.005640  5748 segment.cpp:166] segment_name=0.0.0.0:12795, warn=segment_already_exists
W0706 03:39:19.005652  5748 segment.cpp:253] segment_name=0.0.0.0:12795, warn=segment_already_exists
WARNING: Logging before InitGoogleLogging() is written to STDERR
I0706 03:39:19.995110  6060 client_metric.cpp:114] Client metrics enabled (default enabled)
I0706 03:39:19.995131  6060 client_metric.cpp:115] Client bandwidth summary enabled via MC_STORE_CLIENT_METRIC_BANDWIDTH
I0706 03:39:20.005278  6021 real_client.cpp:5303] dummy_client_monitor_thread stopped
I0706 03:39:21.137727  5740 master_admin_service.cpp:302] Master Admin Metrics: role=leader, state=serving, service_ready=true, master={Mem Storage: 0 B / 0 B | NVMe-oF SSD: 0 B / 0 B | SSD Storage: 0 B / 0 B | Keys: 0 (soft-pinned: 0) | Clients: 4 | Requests (Success/Total per sec): PutStart=0.00/0.00, PutEnd=0.00/0.00, PutRevoke=0.00/0.00, Get=0.20/0.20, Exist=0.00/0.00, Del=0.10/0.10, DelAll=0.00/0.00, Ping=1.10/1.10, CopyStart=0.00/0.00, CopyEnd=0.00/0.00, CopyRevoke=0.00/0.00, MoveStart=0.00/0.00, MoveEnd=0.00/0.00, MoveRevoke=0.00/0.00, EvictDiskReplica=0.00/0.00 | Batch Requests (per sec, Req=Success/PartialSuccess/Total, Item=Success/Total): PutStart:(Req=0.00/0.00/0.00, Item=0.00/0.00), PutEnd:(Req=0.00/0.00/0.00, Item=0.00/0.00), PutRevoke:(Req=0.00/0.00/0.00, Item=0.00/0.00), Get:(Req=0.00/0.00/0.00, Item=0.00/0.00), ExistKey:(Req=0.00/0.00/0.00, Item=0.00/0.00), QueryIp:(Req=0.00/0.00/0.00, Item=0.00/0.00), Clear:(Req=0.00/0.00/0.00, Item=0.00/0.00), CreateMoveTask:(Req=0.00/0.00), CreateCopyTask:(Req=0.00/0.00), QueryTask:(Req=0.00/0.00), FetchTasks:(Req=0.70/0.70), MarkTaskToComplete:(Req=0.00/0.00) | Eviction: Success/Attempts=0/0, AllocFail=0, keys=0, size=0 B | Mem Eviction: Success/Attempts=0/0, keys=0, size=0 B | NoF Eviction: Success/Attempts=0/0, keys=0, size=0 B | Discard: Released/Total=0/0, StagingSize=0 B | Promotion: in_flight=0, admitted=0, completed=0, failed=0, cancelled=0, expired=0, bytes=0 B, rejected(freq/wm/cap)=0/0/0 | Snapshots: Success=0, Fail=0}, ha={HA Metrics Summary: last_seq=0, applied_seq=0, lag=0, pending=0, mutation_queue=0, batch_commits=0, sync_commits=0, skipped=0, checksum_fail=0, etcd_fail=0, watch_disconn=0, state=0}
I0706 03:39:22.118194  5736 master_service.cpp:7837] client_id=14214222416797870286-13568336764865054642, action=client_expired
E0706 03:39:24.265900  6068 dummy_client.cpp:160] Dummy Client not available
E0706 03:39:24.265985  6060 dummy_client.cpp:442] Failed to connect to real client
ERROR

======================================================================
ERROR: setUpClass (__main__.TestDistributedObjectStoreSingleStore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/Mooncake/Mooncake/mooncake-wheel/tests/test_dummy_client.py", line 40, in setUpClass
    get_client(cls.store)
  File "/home/runner/work/Mooncake/Mooncake/mooncake-wheel/tests/test_dummy_client.py", line 31, in get_client
    raise RuntimeError(f"Failed to setup store client. Return code: {retcode}")
RuntimeError: Failed to setup store client. Return code: -1

----------------------------------------------------------------------
Ran 0 tests in 4.272s

FAILED (errors=1)
I0706 03:39:24.270411  6060 dummy_client.cpp:544] [unregister_shm] client_id=5134548976280818569-3214271095265645752
E0706 03:39:24.270424  6060 dummy_client.cpp:146] Dummy Client not connected
I0706 03:39:25.118574  5736 master_service.cpp:7837] client_id=9461847794888464687-13757012156574636724, action=client_expired
I0706 03:39:27.118871  5736 master_service.cpp:7837] client_id=15081876668421406956-11290124828467649169, action=client_expired
Error: Process completed with exit code 1.
```

--- a/mooncake-store/src/real_client_main.cpp
+++ b/mooncake-store/src/real_client_main.cpp
@@ -17,6 +17,7 @@ DEFINE_string(master_server_address, "127.0.0.1:50051",
 DEFINE_string(protocol, "tcp", "Protocol");
 DEFINE_int32(port, 50052, "Real Client service port");
 DEFINE_string(global_segment_size, "4 GB", "Size of global segment");
+DEFINE_string(local_buffer_size, "0", "Size of local buffer (e.g., 16MB, 1GB)");
 DEFINE_int32(threads, 1, "Number of threads for client service");
 DEFINE_string(tenant_id, "default", "Tenant identifier");
 DEFINE_bool(enable_offload, false, "Enable offload availability");
@@ -103,6 +104,7 @@ int main(int argc, char *argv[]) {
     }
 
     size_t global_segment_size = string_to_byte_size(FLAGS_global_segment_size);
+    size_t local_buffer_size = string_to_byte_size(FLAGS_local_buffer_size);
 #ifdef USE_ASCEND_DIRECT
     // just set to true, does not affect GPU process.
     globalConfig().ascend_agent_mode = true;
@@ -110,10 +112,11 @@ int main(int argc, char *argv[]) {
 
     auto client_inst = RealClient::create();
     auto res = client_inst->setup_internal(
-        FLAGS_host, FLAGS_metadata_server, global_segment_size, 0,
-        FLAGS_protocol, FLAGS_device_names, FLAGS_master_server_address,
-        nullptr, "@mooncake_client_" + std::to_string(FLAGS_port) + ".sock",
-        FLAGS_port, FLAGS_enable_offload, FLAGS_start_offload_rpc_server, "",
+        FLAGS_host, FLAGS_metadata_server, global_segment_size,
+        local_buffer_size, FLAGS_protocol, FLAGS_device_names,
+        FLAGS_master_server_address, nullptr,
+        "@mooncake_client_" + std::to_string(FLAGS_port) + ".sock", FLAGS_port,
+        FLAGS_enable_offload, FLAGS_start_offload_rpc_server, "",
         FLAGS_tenant_id);
     if (!res) {
         LOG(FATAL) << "Failed to setup client: " << toString(res.error());

